src/block/block_compact.c

```diff
@@ -170,14 +213,19 @@ __block_compact_estimate_remaining_work(WT_SESSION_IMPL *session, WT_BLOCK *bloc
      * either.
      */
     avg_block_size = (wt_off_t)WT_ALIGN(
-      block->compact_bytes_rewritten / block->compact_pages_rewritten, block->allocsize);
+      block->compact_bytes_reviewed / block->compact_pages_reviewed, block->allocsize);
+
+    /* We don't currently have a way to track the internal page size, but this should be okay. */
+    avg_internal_block_size = 4096;
```

Comment:
Maybe not worth the fuss, but there is a runtime parameter controlling the maximum internal page size that maybe you could use here? (See `internal_page_max` config option to [WT_SESSION::create](http://source.wiredtiger.com/develop/struct_w_t___s_e_s_s_i_o_n.html#a358ca4141d59c345f401c58501276bbb)). It gets loaded into WT_BTREE.maxintlpage and should be reachable via the session.

Reply:
This sounds like a good idea. I looked into this, and so far I haven't found a better way to get `internal_page_max` would be to pull it from the WT_BTREE as you mentioned, which is in the layer above. I could presumably get it from `S2BT(session)`. Just to make sure, would this be okay, or would this be perhaps too much of a layering violation?

Reply:
Yeah, it would be nice to avoid reaching into the btree from the block manager. 

It would probably be simpler/cleaner to just use `block->allocsize`. That's the unit of allocation from the file -- so the minimum block size. It defaults to 4KB, so it's no worse than using the constant here.

If that isn't good enough, we could consider a cleaner way of passing `internal_page_max` to the block manager. Maybe part of doing compaction estimation should be to pass interesting parameters into the block manager? Or maybe there should be a callback to the Btree? Neither seem (to me) to be worth pursuing without some data that what you've got isn't sufficient.

Reply:
Sure, let's use the allocation size for now and revisit these options if we have a reason to believe that the extra accuracy is worth the effort.